src/viking.js

```diff
@@ -1,11 +1,86 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor (health, strength){
+        this.health = health;
+        this.strength = strength
+    };
+    attack (){
+        return this.strength;
+    };
+    receiveDamage (thedamage) {
+        this.thedamage=thedamage;
+        this.health=this.health - this.thedamage;
+    };
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor (name, health, strength){
+        super(health, strength);
+        this.name = name;
+    };
+    receiveDamage (vikingReceivedDamage) {
+        super.receiveDamage (vikingReceivedDamage);
+        if (this.health>0){
+            return `${this.name} has received ${vikingReceivedDamage} points of damage`;
+        }else {
+            return `${this.name} has died in act of combat`;
+
+        }
+    }
+    battleCry (){
+        return "Odin Owns You All!";
+    };
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier{
+    constructor (health, strength){
+        super(health, strength);
+    };
+    receiveDamage (saxonReceivedDamage) {
+        super.receiveDamage (saxonReceivedDamage);
+        if (this.health>0){
+            return `A Saxon has received ${saxonReceivedDamage} points of damage`;
+        }else {
+            return `A Saxon has died in combat`;
+
+        }
+    }
+}
 
 // War
-class War {}
+class War {
+
+    vikingArmy=[];
+    saxonArmy=[];
+    
+    addViking (aViking){
+        this.aViking=aViking;
+        this.vikingArmy.push(this.aViking);
+    }
+    addSaxon (aSaxon){
+        this.aSaxon=aSaxon;
+        this.saxonArmy.push(this.aSaxon);
+    }
+
+    vikingAttack (){
```

Comment:
Aquí teníamos que implementar un random para hacerlos pelear, todos contra todos je,  después de eso y midiendo su healt lo sacamos del arreglo te dejo un posible solución y lo mismito implementas con el ataque del saxon:
 vikingAttack(){
      let saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)]
      let viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)]
     
        if(saxon.health <= viking.strength){
            this.saxonArmy.pop(saxon)
        }
        return saxon.receiveDamage(viking.strength)
    }